### ISSUE TITLE
How to generate multiple ‘extremum’ outliers?

### ISSUE THREAD

**Issue by Bigchicken98** [HPI-Information-Systems/gutentag#1]
![image](https://user-images.githubusercontent.com/26326911/196389268-ee0409c5-2fa5-4155-a807-6b152048d039.png)

**Comment by CodeLionX** [comment]
Hey,

You are trying to generate an extremum anomaly of length 15. This is not allowed because extreme values just consist of a single point. If you want to generate multiple extremum anomalies, just add multiple anomaly definitions:

```yaml
timeseries:
- name: demo
  length: 1000
  base-oscillations:
  - kind: sine
    frequency: 4.0
    amplitude: 1.0
    variance: 0.05
  anomalies:
  # 1. extremum anomaly
  - position: beginning
    length: 1
    kinds:
    - kind: extremum
      min: true
      local: false
  # 2. extremum anomaly
  - &extremum-anomaly
    position: middle
    length: 1
    kinds:
    - kind: extremum
      min: false
      local: true
      context_window: 50
  # 3. extremum anomaly
  - <<: *extremum-anomaly
  # ...
```

![Figure](https://user-images.githubusercontent.com/10573700/196448272-121e1aa5-20f0-4749-9d10-f235943cddd8.png)

**Comment by Bigchicken98** [comment]
I adopted your method. I wanted to generate 50 extremum anomalies in base oscillation with length of 1000, but only 32 were generated

**Comment by CodeLionX** [comment]
50 extreme values in a time series of length 1000 is a contamination of 5%. Don't you think that this is too much for an anomaly detection use case?

Nevertheless, there is only so much that the automatic positioning algorithm in GutenTAG can do. With so many anomalies, you should specify their location manually and don't rely on the "beginning", "middle", "end" positions. You can do this with the `extact-positiong`-key:

```yaml
timeseries:
- name: demo
  length: 1000
  base-oscillations:
  - kind: sine
    frequency: 4.0
    amplitude: 1.0
    variance: 0.05
  anomalies:
  - exact-position: 413  # index beginning from 0
    length: 1
    kinds:
    - kind: extremum
      min: true
      local: false
  # ...
```

---

I want to have a look why the anomalies disappear. Can you send me your configuration file?

**Comment by Bigchicken98** [comment]
[timeseries-declarations.txt](https://github.com/HPI-Information-Systems/gutentag/files/9825749/timeseries-declarations.txt)
Thank you!

**Comment by CodeLionX** [comment]
> [timeseries-declarations.txt](https://github.com/HPI-Information-Systems/gutentag/files/9825749/timeseries-declarations.txt) Thank you!

There is actually a bug in the positioning algorithm that leads to multiple anomalies getting injected at the same position. We are working on a fix for that.

**Comment by CodeLionX** [comment]
We fixed the positioning algorithm in 801b0a316f2a829f0c2d539006cc6a12bb6bba82. GutenTAG should not miss any further anomalies from now on. You can use the current `master` to test this.

**Comment by Bigchicken98** [comment]
Thank you for your help
